lib/src/core/comms/secure_channel.rs

```diff
@@ -35,6 +36,12 @@ pub enum Role {
     Server,
 }
 
+#[derive(Debug)]
+struct RemoteKeys {
+    keys: (Vec<u8>, AesKey, Vec<u8>),
+    expires_at: DateTime,
```

Comment:
Avoid using `DateTime` for timestamps, if possible, confusing as that may sound.

If time is purely relative, always prefer using `std::time::Instant` and `std::time::Duration`, these generally point to the monotonic high frequency clock, while clock time like `SystemTime` or those provided by `chrono` use the system clock, which can suddenly change for a number of reasons.

Reply:
Fair point, will change it to `Instance`

Reply:
Hmm... After taking a closer look I don't think that makes sense here. The expiration time is calculated based on the time the token was created. That creation time comes from the server in the form of a timestamp and is most likely in the past by the time we get here.

So somehow converting that to an `Instant` doesn't make much sense to me especially taking into account that there are lots and lots of other places in the code that do use a `DateTime`. Which in most cases makes sence because these values are sometimes also used in messages between the client and the server.

Of course we could so something like the pseudo code pasted below, but I honestly don't see much benefit from that:

```rust
let expires_in = Instant::now() + (DateTime::now() - (self.token_created_at + expires_in));
```

While the argument about the system clock if very true, in the case of this crate a whole lot more will break (for example the secure channel) if the system clock got updated while the program is running. So unless you are thinking about another solution or have more insights to share, I think I prefer to keep it like this.

Reply:
I didn't recall that it was received from the server, doing it that way is _usually_ suboptimal, but OPC-UA isn't always optimal... If so then you're right, we need to use DateTime, with all its potential issues.